From ade59c4fededa71289fc51b3f810cbd7fae4b08b Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 5 Apr 2024 10:12:24 +0200
Subject: [PATCH 0810/1695] drivers: stm32_tamp: update handling of TAMP wakeup
 EXTI line

The TAMP secure EXTI line does not have the same ID for different
platforms. Move this information in the compatible data.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f56a332f9796255e7fdf656b5d2da5061576712a drivers: stm32_tamp: update handling of TAMP wakeup EXTI line]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ie48ff039bada807fe7e1173fb8a68afc022bd735
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/371583
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_tamp.c         | 9 ++++++---
 core/include/drivers/stm32_tamp.h | 1 +
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index aebf7131e..0bfc7b72e 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -235,8 +235,6 @@
 #define TAMP_ACTIVE			BIT(17)
 #define TAMP_IN_DT			BIT(18)
 
-#define TAMP_EXTI_WKUP			U(18)
-
 /*
  * RIF miscellaneous
  */
@@ -2085,9 +2083,11 @@ static TEE_Result stm32_tamp_probe(const void *fdt, int node,
 		goto err;
 
 	if (stm32_tamp.pdata.is_wakeup_source) {
+		struct stm32_tamp_compat *compat = stm32_tamp.pdata.compat;
+
 		if (IS_ENABLED(CFG_STM32_EXTI))
 			stm32_exti_enable_wake(stm32_tamp.pdata.exti,
-					       TAMP_EXTI_WKUP);
+					       compat->exti_wakeup_line);
 		else
 			DMSG("TAMP event are not configured as wakeup source");
 	}
@@ -2143,6 +2143,7 @@ static const struct stm32_tamp_compat mp13_compat = {
 		.ext_tamp_size = ARRAY_SIZE(ext_tamp_mp13),
 		.pin_map = pin_map_mp13,
 		.pin_map_size = ARRAY_SIZE(pin_map_mp13),
+		.exti_wakeup_line = U(18),
 };
 
 static const struct stm32_tamp_compat mp15_compat = {
@@ -2154,6 +2155,7 @@ static const struct stm32_tamp_compat mp15_compat = {
 		.ext_tamp_size = ARRAY_SIZE(ext_tamp_mp15),
 		.pin_map = pin_map_mp15,
 		.pin_map_size = ARRAY_SIZE(pin_map_mp15),
+		.exti_wakeup_line = U(18),
 };
 
 static const struct stm32_tamp_compat mp25_compat = {
@@ -2171,6 +2173,7 @@ static const struct stm32_tamp_compat mp25_compat = {
 		.ext_tamp_size = ARRAY_SIZE(ext_tamp_mp25),
 		.pin_map = pin_map_mp25,
 		.pin_map_size = ARRAY_SIZE(pin_map_mp25),
+		.exti_wakeup_line = U(21),
 };
 
 static const struct stm32_tamp_compat mp23_compat = {
diff --git a/core/include/drivers/stm32_tamp.h b/core/include/drivers/stm32_tamp.h
index 61aa7f5e5..9601ae383 100644
--- a/core/include/drivers/stm32_tamp.h
+++ b/core/include/drivers/stm32_tamp.h
@@ -109,6 +109,7 @@ struct stm32_tamp_compat {
 	uint32_t int_tamp_size;
 	uint32_t pin_map_size;
 	uint32_t tags;
+	uint32_t exti_wakeup_line;
 };
 
 struct stm32_tamp_platdata {
-- 
2.47.2

